File: src/main/java/org/steamproject/model/Purchase.java
package org.steamproject.model;

/**
 * Représente un achat effectué par un joueur.
 */
public class Purchase {
    private String purchaseId;
    private String gameId;
    private String gameName;
    private String playerId;
    private String playerUsername;
    private double pricePaid;
    private String platform;
    private long timestamp;
    private boolean isDlc;
    private String dlcId; // null si achat de jeu principal

    public Purchase() {}

    public Purchase(String purchaseId, String gameId, String gameName, String playerId, 
                    String playerUsername, double pricePaid, String platform, long timestamp) {
        this.purchaseId = purchaseId;
        this.gameId = gameId;
        this.gameName = gameName;
        this.playerId = playerId;
        this.playerUsername = playerUsername;
        this.pricePaid = pricePaid;
        this.platform = platform;
        this.timestamp = timestamp;
        this.isDlc = false;
        this.dlcId = null;
    }

    // Getters & Setters
    public String getPurchaseId() { return purchaseId; }
    public void setPurchaseId(String purchaseId) { this.purchaseId = purchaseId; }

    public String getGameId() { return gameId; }
    public void setGameId(String gameId) { this.gameId = gameId; }

    public String getGameName() { return gameName; }
    public void setGameName(String gameName) { this.gameName = gameName; }

    public String getPlayerId() { return playerId; }
    public void setPlayerId(String playerId) { this.playerId = playerId; }

    public String getPlayerUsername() { return playerUsername; }
    public void setPlayerUsername(String playerUsername) { this.playerUsername = playerUsername; }

    public double getPricePaid() { return pricePaid; }
    public void setPricePaid(double pricePaid) { this.pricePaid = pricePaid; }

    public String getPlatform() { return platform; }
    public void setPlatform(String platform) { this.platform = platform; }

    public long getTimestamp() { return timestamp; }
    public void setTimestamp(long timestamp) { this.timestamp = timestamp; }

    public boolean isDlc() { return isDlc; }
    public void setDlc(boolean dlc) { isDlc = dlc; }

    public String getDlcId() { return dlcId; }
    public void setDlcId(String dlcId) { this.dlcId = dlcId; }

    @Override
    public String toString() {
        return "Purchase{" +
                "purchaseId='" + purchaseId + '\'' +
                ", gameId='" + gameId + '\'' +
                ", gameName='" + gameName + '\'' +
                ", playerId='" + playerId + '\'' +
                ", pricePaid=" + pricePaid +
                ", platform='" + platform + '\'' +
                ", isDlc=" + isDlc +
                '}';
    }
}

File: src/main/kotlin/model/Purchase.kt
package org.example.model

/**
 * Représente un achat effectué par un joueur.
 */
data class Purchase(
    val purchaseId: String,
    val gameId: String,
    val gameName: String,
    val playerId: String,
    val playerUsername: String,
    val pricePaid: Double,
    val platform: String,
    val timestamp: Long,
    val isDlc: Boolean = false,
    val dlcId: String? = null
)

/**
 * Représente un DLC (contenu téléchargeable).
 */
data class DLC(
    val id: String,
    val parentGameId: String,
    val name: String,
    val description: String,
    val price: Double,
    val minGameVersion: String?, // Version minimale requise
    val releaseDate: String,
    val sizeInMB: Long,
    val standalone: Boolean = false // Peut être joué sans le jeu principal
)

/**
 * Facteurs de prix pour l'algorithme qualité/demande.
 */
data class PriceFactors(
    val basePrice: Double,
    val qualityFactor: Double, // Basé sur les ratings (0.8 - 1.2)
    val demandFactor: Double,  // Basé sur les ventes (0.9 - 1.3)
    val promotionFactor: Double, // Réduction promotionnelle
    val finalPrice: Double
) {
    val isOnPromotion: Boolean get() = promotionFactor < 1.0
    val promotionPercent: Double get() = if (isOnPromotion) (1.0 - promotionFactor) * 100.0 else 0.0
}

/**
 * Statistiques d'achats.
 */
data class PurchaseStats(
    val totalPurchases: Int,
    val totalRevenue: Double
)
